app/src/src_nonfreenet/org/breezyweather/sources/meteoam/MeteoAmResultConverter.kt

```diff
@@ -0,0 +1,213 @@
+            dailyForecast.add(
+                Daily(
+                    date = formatter.parse(it.localDate)!!,
+                    day = HalfDay(
+                        weatherText = getWeatherText(it.icon),
+                        weatherCode = getWeatherCode(it.icon)
+                    ),
+                    night = HalfDay(
+                        weatherText = getWeatherText(it.icon),
+                        weatherCode = getWeatherCode(it.icon)
+                    )
+                )
+            )
+        }
+    }
+    return dailyForecast
+}
+
+private fun getHourlyForecast(
+    timeseries: List<String>,
+    data: MeteoAmForecastData
+): List<HourlyWrapper> {
+    val formatter = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssX", Locale.ITALIAN)
+    formatter.timeZone = TimeZone.getTimeZone("Europe/Rome")
+    val hourlyForecast = mutableListOf<HourlyWrapper>()
+    for (i in timeseries.indices) {
+        hourlyForecast.add(
+            HourlyWrapper(
+                date = formatter.parse(timeseries[i])!!,
+                weatherText = getWeatherText(data.icon?.get(i.toString())),
+                weatherCode = getWeatherCode(data.icon?.get(i.toString())),
+                temperature = Temperature(
+                    temperature = data.temp?.get(i.toString())?.toDouble()
+                ),
+                precipitationProbability = PrecipitationProbability(
+                    total = data.tpop?.get(i.toString())?.toDouble()
+                ),
+                wind = Wind(
+                    degree = data.wdir?.get(i.toString())?.toDouble(),
+                    speed = data.wkph?.get(i.toString())?.toDouble()?.div(3.6)
+                ),
+                relativeHumidity = data.rhum?.get(i.toString())?.toDouble(),
+                pressure = data.pres?.get(i.toString())?.toDouble()
+            )
+        )
+    }
+    return hourlyForecast
+}
+
+// Icon definitions can be found at https://www.meteoam.it/it/legenda-simboli
+// Icon sources can be found at https://www-static.meteoam.it/maps/img/icon_web_v4/{icon}.svg
+// Icons 06 and 07 are duplicates
+private fun getWeatherCode(icon: String?): WeatherCode? {
+    return when (icon) {
+        "01", "31" -> WeatherCode.CLEAR
+        "02", "03", "04", "32", "33", "34" -> WeatherCode.PARTLY_CLOUDY
+        "05", "06", "07", "35" -> WeatherCode.CLOUDY
+        "08", "09" -> WeatherCode.RAIN
+        "10" -> WeatherCode.THUNDER
+        "11", "12" -> WeatherCode.SLEET
+        "13", "14", "36" -> WeatherCode.FOG
+        "15" -> WeatherCode.HAIL
+        "16" -> WeatherCode.SNOW
+        "17", "19" -> WeatherCode.WIND
+        "18" -> WeatherCode.HAZE
+        else -> null
+    }
+}
+
+// Icon definitions can be found at https://www.meteoam.it/it/legenda-simboli
+// Icon sources can be found at https://www-static.meteoam.it/maps/img/icon_web_v4/{icon}.svg
+// Icons 06 and 07 are duplicates
+private fun getWeatherText(icon: String?): String? {
```

Comment:
Do you think you could match common translations?

Implementation: https://github.com/breezy-weather/breezy-weather/blob/v5.2.5/app/src/main/java/org/breezyweather/sources/openmeteo/OpenMeteoResultConverter.kt#L247

Available `common` translations: https://github.com/breezy-weather/breezy-weather/blob/v5.2.5/app/src/main/res/values/strings.xml#L751-L795

Reply:
Matched most of them. Here are the outstanding ones:
| Condition | Description | Comment |
|-|-|-|
| Parzialmente Velato | Nubi alte e molto sottili | _Parzialmente Velato_ translates to "partially veiled" and can be considered "mostly clear", however… |
| Velato | Nubi traslucide che coprono totalmente il cielo | _Velato_ ("veiled") can also be considered "mostly clear". There must be a reason for the two separate conditions. Maybe it's "partly clear"? |
| Temporale | Presenza contestuale di precipitazione, fulminazioni e tuoni | "Thunderstorm", not found in `common_weather_text` but there is a string labeled `weather_kind_thunderstorm`. |
| Foschia | Cielo opaco e riduzione della visibilità | "Haze", not found in `common_weather_text` but there is a string labeled `weather_kind_haze`. |
| Grandine | Precipitazione solida in grani di ghiaccio | "Hail", not found in `common_weather_text` but there is a string labeled `weather_kind_hail`. |
| Tromba d’aria o marina | Vento vorticoso molto intenso con sollevamento di acqua, polvere e oggetti | "Tornado or waterspout" |
| Fumo | Visibilità ridotta per presenza di fumo | "Smoke" |
| Tempesta di sabbia | Vento intenso con sollevamento di sabbia | "Sand storm" |

Full list can be found [here](https://www.meteoam.it/it/legenda-simboli).